File: go/gof/template_method/concrete_y.go
package template_method

import (
	"fmt"
)

type siteYCrawler struct {
	name string
	path string
}

func NewSiteYCrawler(name, path string) *siteYCrawler {
	return &siteYCrawler{
		name: name,
		path: path,
	}
}

func (x *siteYCrawler) Name() string {
	return x.name
}

func (x *siteYCrawler) Get() error {
	fmt.Printf("Get request: %s\n", x.path)
	return nil
}

func (x *siteYCrawler) Scrape() error {
	fmt.Println("Scraping now.")
	fmt.Println("Scraping now..")
	fmt.Println("Scraping now...")
	return nil
}

func (x *siteYCrawler) Store() error {
	fmt.Println("Stored!")
	return nil
}

File: python/gof/template_method/concrete_y.py
from abstract import AbstractCrawler

class SiteYCrawler(AbstractCrawler):
    def __init__(self, name:str, path:str):
        self.__name = name
        self.__path = path

    def name(self) -> str:
        return self.__name

    def get(self):
        print('Get request: {}'.format(self.__path))

    def scrape(self):
        print('Scraping now.')
        print('Scraping now..')
        print('Scraping now...')

    def store(self):
        print('Stored!')